### ISSUE TITLE
Learning Rate lowering to 0.000002

### ISSUE THREAD

**Issue by ChaosDeity123** [bmaltais/kohya_ss#3262]
I tried training a LoRA with Prodigy optimizer (LR:1, UNet:1 and TE:0), after the training I saw LR:0.000002 or 0.000001 using LoRA inspector. 

I tried training with AdamW and AdamW8bit also but I keep getting same the problem.

Why is this happening? I need some help.

**Comment by ChaosDeity123** [comment]
[Tofuuu_20250430-005307.json](https://github.com/user-attachments/files/20632169/Tofuuu_20250430-005307.json)
This is my LoRA's Json. It shows LR:1 here but after the training it goes 0.000002 or 0.000001.
> I might have found the issue. Can you try the fix I posted in a new branch:
> 
> git fetch git checkout fix-lora-lr-handling


> This is frustrating. Let's get to the bottom of this issue. Can you share the JSON code you use so I reproduce the same training condition to observe the root cause?

**Comment by ChaosDeity123** [comment]
"modelspec.implementation": "https://github.com/Stability-AI/generative-models",
  "ss_debiased_estimation": "False",
  "modelspec.architecture": "stable-diffusion-xl-v1-base/lora",
  "modelspec.encoder_layer": 2,
  "ss_validate_every_n_epochs": "None",
  "ss_min_snr_gamma": 5,
  "ss_huber_c": 0.1,
  "ss_max_validation_steps": "None",
  "ss_resize_interpolation": "None",
  "sshs_model_hash": "7b1fc93af43a9aaab54dbb318a0545b59aca90ed376f72894ba00eca3dc01f57",
  "ss_batch_size_per_device": 3,
  "modelspec.date": "2025-05-30T03:17:10",
  "ss_zero_terminal_snr": "False",
  "ss_reg_dataset_dirs": {},
  "ss_sd_model_name": "illustriousXL_v01.safetensors",
  "ss_num_reg_images": 0,
  "ss_huber_scale": 1,
  "ss_num_validation_images": 0,
  "ss_num_epochs": 10,
  "ss_seed": 3528184257,
  "ss_noise_offset_random_strength": "False",
  "ss_validation_split": 0,
  "ss_vae_hash": "d636e597",
  "modelspec.sai_model_spec": "1.0.0",
  "ss_keep_tokens": 0,
  "modelspec.title": "Cstyle",
  "ss_gradient_accumulation_steps": 1,
  "ss_network_alpha": 8,
  "ss_validation_seed": "None",
  "ss_v2": "False",
  "ss_full_fp16": "False",
  "ss_sd_scripts_commit_hash": "5753b8ff6bc045c27c1c61535e35195da860269c",
  "ss_fp8_base_unet": "False",
  "ss_lr_scheduler": "cosine",
  "ss_unet_lr": 1,
  "ss_network_dim": 8,
  "ss_max_grad_norm": 1,
  "ss_optimizer": "prodigyopt.prodigy.Prodigy(decouple=True,weight_decay=0.5,betas=(0.9, 0.99),use_bias_correction=False)",
  "ss_huber_schedule": "snr",
  "ss_noise_offset": 0.05,
  "ss_validate_every_n_steps": "None",
  "ss_resolution": "(1024, 1024)",
  "ss_max_bucket_reso": 2048,

**Comment by ChaosDeity123** [comment]
> `git checkout dev`
> 
> This command solved my problem. I'll stay in dev until the next merge.

Hii may I know how you fix it?
